source/common/http/conn_manager_utility.cc

```diff
@@ -131,6 +131,70 @@ void ConnectionManagerUtility::mutateRequestHeaders(Http::HeaderMap& request_hea
   if (config.tracingConfig()) {
     Tracing::HttpTracerUtility::mutateHeaders(request_headers, runtime);
   }
+  mutateXfccRequestHeader(request_headers, connection, config);
+}
+
+void ConnectionManagerUtility::mutateXfccRequestHeader(Http::HeaderMap& request_headers,
+                                                       Network::Connection& connection,
+                                                       ConnectionManagerConfig& config) {
+  if (config.forwardClientCert() != Http::ForwardClientCertType::AlwaysForwardOnly &&
+      !(connection.ssl() && connection.ssl()->peerCertificatePresented())) {
+    request_headers.removeForwardedClientCert();
+    return;
+  }
+  std::string clientCertDetails;
+  if (config.forwardClientCert() == Http::ForwardClientCertType::AppendForward ||
+      config.forwardClientCert() == Http::ForwardClientCertType::SanitizeSet) {
+    // In these cases, the client cert in the current hop should be set into the XFCC header.
+    if (!connection.ssl()->uriSanLocalCertificate().empty()) {
+      clientCertDetails += "By=" + connection.ssl()->uriSanLocalCertificate();
+    }
+    if (!connection.ssl()->sha256PeerCertificateDigest().empty()) {
+      if (!clientCertDetails.empty()) {
+        clientCertDetails += ";";
```

Comment:
Why not just build a list of clientCertDetails components and then do a https://github.com/lyft/envoy/blob/master/source/common/common/utility.h#L140 with ";"? A join is what's happening logically here.